### PR TITLE
Homework08

### PR DESCRIPTION
**Pull request by YeaaaaH**

### CODE REVIEW

File: HW/src/main/java/homework05/MyHashMap.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * HashMap implementation
+ * @param <K> key
+ * @param <V> value
+ */
+public class MyHashMap<K, V> {
```
Comment (Siril-O) [inline HW/src/main/java/homework05/MyHashMap.java]: When you will learn how to commit only related to homework  files ?

Comment (Siril-O) [reply HW/src/main/java/homework05/MyHashMap.java]: It is  must have skill in production

File: HW/src/main/java/homework08/task1/MailService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class MailService<T> implements Consumer<Consumerable<T>> {
+
+    private Map<String, List<T>> mailBox;
+
+    public Map<String, List<T>> getMailBox() {
+        return mailBox;
+    }
+
+    @Override
+    public void accept(Consumerable<T> tConsumerable) {
```
Comment (Siril-O) [inline HW/src/main/java/homework08/task1/MailService.java]: Why it is empty? Add your logic

File: HW/src/main/java/homework08/task3/RandomStream.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package homework08.task3;
+
+import java.util.stream.IntStream;
+
+public class RandomStream {
+    public static IntStream pseudoRandStream(int seed) {
+        return  IntStream.iterate(seed, n -> n * n / 10 % 1000);
```
Comment (Siril-O) [inline HW/src/main/java/homework08/task3/RandomStream.java]: you should take 2,3,4  digit of number

File: HW/src/main/java/homework08/task2/MinMaxFinding.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+public class MinMaxFinding {
+    public static <T> void findMinMax(
+            Stream<? extends T> stream,
+            Comparator<? super T> order,
+            BiConsumer<? super T, ? super T> minMaxConsumer) {
+        if (stream.equals(Stream.empty())) {
+            minMaxConsumer.accept(null, null);
+        } else {
+            List<T> list1 = stream.collect(Collectors.toList());
```
Comment (Siril-O) [inline HW/src/main/java/homework08/task2/MinMaxFinding.java]: why you create lists form stream and again stream from list?

File: HW/src/main/java/homework08/task1/Consumerable.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package homework08.task1;
+
+public interface Consumerable<T> {
```
Comment (Siril-O) [inline HW/src/main/java/homework08/task1/Consumerable.java]: Bad name

File: HW/src/main/java/homework08/task1/Consumerable.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    String getFrom();
+    String getTo();
```
Comment (Siril-O) [inline HW/src/main/java/homework08/task1/Consumerable.java]: add method T content()

File: HW/src/main/java/homework08/task3/RandomStream.java (hunk 3 above)
Comment (YeaaaaH) [reply HW/src/main/java/homework08/task3/RandomStream.java]: But it accepts this solution

File: HW/src/main/java/homework08/task2/MinMaxFinding.java (hunk 4 above)
Comment (YeaaaaH) [reply HW/src/main/java/homework08/task2/MinMaxFinding.java]: To separate min and max values

**Pull request merged**